client/src/main/java/org/spine3/protobuf/ProtoJavaMapper.java

```diff
@@ -0,0 +1,366 @@
+            checkState(input instanceof Message);
+            final Message result = (Message) input;
+            return result;
+        }
+    }
+
+    private static class PrimitiveTypeCaster<T> extends AnyCaster<T> {
+
+        private static final ImmutableMap<Class, Converter<? extends Message, ?>>
+                PROTO_WRAPPER_TO_HANDLER =
+                ImmutableMap.<Class, Converter<? extends Message, ?>>builder()
+                        .put(Int32Value.class, new Int32Handler())
+                        .put(Int64Value.class, new Int64Handler())
+                        .put(UInt32Value.class, new UInt32Handler())
+                        .put(UInt64Value.class, new UInt64Handler())
+                        .put(FloatValue.class, new FloatHandler())
+                        .put(DoubleValue.class, new DoubleHandler())
+                        .put(BoolValue.class, new BoolHandler())
+                        .put(StringValue.class, new StringHandler())
+                        .build();
+        private static final ImmutableMap<Class, Converter<? extends Message, ?>>
+                PRIMITIVE_TO_HANDLER =
+                ImmutableMap.<Class, Converter<? extends Message, ?>>builder()
+                        .put(Integer.class, new Int32Handler())
+                        .put(Long.class, new Int64Handler())
+                        .put(Float.class, new FloatHandler())
+                        .put(Double.class, new DoubleHandler())
+                        .put(Boolean.class, new BoolHandler())
+                        .put(String.class, new StringHandler())
+                        .build();
+
+        @Override
+        protected T cast(Any input) {
+            final Message unpacked = AnyPacker.unpack(input);
+            final Class boxedType = unpacked.getClass();
+            @SuppressWarnings("unchecked") final Function<Message, T> typeUnpacker =
+                    (Function<Message, T>) PROTO_WRAPPER_TO_HANDLER.get(boxedType);
+            checkArgument(typeUnpacker != null,
+                          "Could not find a primitive type for %s.",
+                          boxedType.getCanonicalName());
+            final T result = typeUnpacker.apply(unpacked);
+            return result;
+        }
+
+        @Override
+        protected Message uncast(T input) {
+            final Class<?> cls = input.getClass();
+            @SuppressWarnings("unchecked") final Converter<Message, T> converter =
+                    (Converter<Message, T>) PRIMITIVE_TO_HANDLER.get(cls);
+            checkArgument(converter != null,
+                          "Could not find a wrapper type for %s.",
+                          cls.getCanonicalName());
+            final Message result = converter.reverse()
+                                            .convert(input);
+            return result;
+        }
+    }
+
+    abstract static class PrimitiveHandler<M extends Message, T> extends Converter<M, T> {
```

Comment:
Requires documentation.

Also, it is declared as `package-private`, but I don't see why it cannot be `private`. Perhaps, `@VisibleForTesting` is missing?